src/main/java/com/ebivariation/contigalias/controller/authentication/SecurityConfiguration.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ebivariation.contigalias.controller.authentication;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.builders.WebSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfiguration extends WebSecurityConfigurerAdapter {
+
+    private static final String REALM = "EBI-REALM";
+
+    private static final String ROLE_ADMIN = "ADMIN";
+
+    @Value("${controller.auth.admin.username}")
+    private String USERNAME_ADMIN;
+
+    @Value("${controller.auth.admin.password}")
+    private String PASSWORD_ADMIN;
+
+    @Autowired
+    public void configureGlobalSecurity(AuthenticationManagerBuilder auth) throws Exception {
+        auth.inMemoryAuthentication().withUser(USERNAME_ADMIN).password("{noop}" + PASSWORD_ADMIN).roles(ROLE_ADMIN);
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http.csrf().disable()
+            .authorizeRequests()
+            .antMatchers("/contig-alias/**").permitAll()
+            .antMatchers("/contig-alias-admin/**").hasRole(ROLE_ADMIN)
+            .and().httpBasic().realmName(REALM)
+            .authenticationEntryPoint(getBasicAuthEntryPoint())
+            .and().sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS);
+    }
+
+    @Bean
+    public CustomBasicAuthenticationEntryPoint getBasicAuthEntryPoint() {
+        return new CustomBasicAuthenticationEntryPoint();
```

Comment:
I think injecting the Bean into this class would be better than instantiating it.

You need to declare it and receive it in the constructor
```    
private CustomBasicAuthenticationEntryPoint customBasicAuthenticationEntryPoint;

public SecurityConfiguration(CustomBasicAuthenticationEntryPoint customBasicAuthenticationEntryPoint) {
        this.customBasicAuthenticationEntryPoint = customBasicAuthenticationEntryPoint;
    }
```
And then use it instead of calling the method (that should be deleted)
Note that this change will require the `CustomBasicAuthenticationEntryPoint` class to be annotated with `@Component`

Reply:
```
Description:

Parameter 0 of constructor in com.ebivariation.contigalias.controller.authentication.SecurityConfiguration required a bean of type 'com.ebivariation.contigalias.controller.authentication.CustomBasicAuthenticationEntryPoint' that could not be found.


Action:

Consider defining a bean of type 'com.ebivariation.contigalias.controller.authentication.CustomBasicAuthenticationEntryPoint' in your configuration.

```

Reply:
My code:- 

```

    private final CustomBasicAuthenticationEntryPoint customBasicAuthenticationEntryPoint;

    @Autowired
    public SecurityConfiguration(CustomBasicAuthenticationEntryPoint customBasicAuthenticationEntryPoint) {
        this.customBasicAuthenticationEntryPoint = customBasicAuthenticationEntryPoint;
    }

```

Reply:
the problem was that the tests don't auto scan components as the main code does. a simple `@Import` fixes it.